lib/screens/user/data_entry_page.dart

```diff
@@ -103,19 +118,7 @@ class _DataEntryPageState extends State<DataEntryPage> {
         sessionManager.setUserId(userId);
         print("UserID: $userId");
 
-        await firestore.collection('users').doc(userId).update({
-          'name': name,
-          'surname': surname,
-          'email': email,
-          'unique_citizens_id': uniqueCitizensId,
-          'date_of_birth': dateOfBirth,
-          'address': address,
-          'city': city,
-          'phone_number': phoneNumber,
-          'blood_type': bloodType?.toString().split('.').last,
-          'gender': gender,
-          'is_first_login': false,
-        });
+        _userDataService.updateUser(userData);
       }
 
       print('Data successfully saved to Firestore.');
```

Comment:
maybe remove this, or add it as a log?

Reply:
I'll definitely check out some Flutter logging frameworks